mlir/lib/Dialect/Bufferization/Transforms/Bufferize.cpp

```diff
@@ -222,6 +222,13 @@ struct OneShotBufferizePass
             [](TensorType t) -> std::optional<Attribute> {
           return std::nullopt;
         };
+      } else if (useEncodingForMemorySpace) {
```

Comment:
Can you emit an error/assertion if both `mustInferMemorySpace` and `useEncodingForMemorySpace` are set?